apis/models/cartDao.js

```diff
@@ -0,0 +1,205 @@
+const database = require('../models/dataSource');
+const { affectedRowsErrorHandler } = require('../utils/error');
+
+//이미지 필요, product_option 필요, stock 필요
+const getCartByUserId = async (userId) => {
+  const cart = await database.query(
+    `
+    SELECT
+        c.id AS cartId,
+        c.user_id AS userId,
+        p.style_code AS styleCode,
+        c.quantity,
+        c.product_option_id AS productOptionId,
+        p.id AS productId,
+        p.name AS productName,
+        po.size_id AS sizeId,
+        s.foot_size AS size,
+        po.stock,
+        p.retail_price AS retailPrice,
+        p.discount_price AS discountPrice,
+        p.thumbnail
+    FROM 
+        carts c
+    JOIN
+        product_options po
+    ON
+        po.id = c.product_option_id
+    JOIN
+        products p
+    ON 
+        po.product_id = p.id
+    JOIN 
+        sizes s
+    ON 
+        po.size_id = s.id
+    WHERE 
+        c.user_id = ?
+    `,
+    [userId]
+  );
+  return cart;
```

Comment:
다음과 같이 고치는 것이 가독성 측면에서 조금 더 보기 좋아 보입니다.
```suggestion
    SELECT
        c.id AS cartId,
        c.user_id AS userId,
        p.style_code AS styleCode,
        c.quantity,
        c.product_option_id AS productOptionId,
        p.id AS productId,
        p.name AS productName,
        po.size_id AS sizeId,
        s.foot_size AS size,
        po.stock,
        p.retail_price AS retailPrice,
        p.discount_price AS discountPrice,
        p.thumbnail
    FROM carts c
    JOIN product_options po ON po.id = c.product_option_id
    JOIN products p ON po.product_id = p.id
    JOIN sizes s ON po.size_id = s.id
    WHERE  c.user_id = ?
    `,
    [userId]
  );
  return cart;
```

Reply:
수정완료